#!/usr/bin/env python
# SLAW OBJECT DETECTION

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t, str_t
#from local_planner_limits import add_generic_localplanner_params

gen = ParameterGenerator()

gen.add("save", bool_t, 0, "height of camera", False)

gen.add("hough_rho",     double_t, 0, "", 1.0, 1.0, 640)
gen.add("hough_theta",   double_t,    0, "", 0.02, 0, 7)
gen.add("hough_thresh", double_t, 0, "", 50, 0, 2000)
gen.add("hough_minLenghth",   double_t, 0, "", 10, 1, 640)
gen.add("hough_maxLineGap",   double_t, 0, "", 10, 1, 2000)
gen.add("canny_high",   double_t, 0, "", 100, 1, 2000)
gen.add("min_v20",   int_t, 0, "", 11, 0, 200)

gen.add("mask_x",       int_t,    0, "", 150, 0, 640)
gen.add("mask_y",       int_t,    0, "", 106, 0, 480)
gen.add("mask_width",   int_t,    0, "", 343, 0, 640)
gen.add("mask_height",  int_t,    0, "", 309, 0, 480)

exit(gen.generate("slaw_object_detection", "HoughLine", "HoughLine"))
